cEP-0024.md

```diff
@@ -0,0 +1,255 @@
+
+- WIP label conditions:
+  1. Reviewer requests changes
+  2. GitMate's coala plugin finds defects
+  3. CI tests fail
+- Approved label conditions:
+  1. Reviewer approves changes
+
+coala can configure this plugin as:
+
+**WIP Label:** `process/WIP`
+
+**Approved label:** `status/approved`
+
+### Fixes vs Closes automated checking
+
+Newcomers often confuse when they should use `Fixes` or `Closes` in a commit
+message. So, when a PR is opened or updated, GitMate should automatically check
+every commit for issues which are being `Fixed` or `Closed`, if those issues
+have a label (configurable) which represents bug, `Fixes` should be used,
+else `Closes` should be used.
+
+When the commit message contains 'Fixes', but referenced issue doesn't have
+label (configurable) which represents bug, then a comment (configurable) will
+be added on the PR. Message to be commented can be configured by coala as:
+
+```
+`Fixes` is used in commit message, but referenced issue doesn't have a
+`type/bug` label, if issue is updated to include the `type/bug` label, then
+ask a maintainer to add required label, else use `Closes`.
+```
+
+When the commit message contains 'Closes', but referenced issue has bug label,
+then a comment (configurable) will added on the PR. Message to be commented
+can be configured by coala as:
+
+```
+`Closes` is used, but referenced issue has a `type/bug` label, if issue is
+updated to remove `type/bug` label, then ask a maintainer to remove required
+label, else use `Fixes`.
+```
+
+In both cases, 'process/pending review' label will be removed and 'process/wip'
+label will be added. This will be part of `Assign label as per review state`
+plugin.
+
+### Assign Plugin
+
+When a user comments on an issue requesting for assignment to the issue,
+GitMate will assign comment author to the issue, if he/she is eligible to get
+assigned, else missing eligibility conditions will be specified as a comment,
+which isn't configurable.
+
+Assign plugin will support configurations for limiting number of certain
+labelled issues a member can work on, block assignment for certain users or
+certain labelled issues, difficulty level order to be followed while assigning
+issues to members so that they should work on lower difficulty level issues
+first.
+
+[corobo](https://github.com/coala/corobo) does many of these things, but it is
+based just on Gitter messages and is used only by coala, but GitMate is based
+on GitHub or GitLab, used by many other users and is configurable.
+
+**Syntax for assign command**: `@gitmate-bot assign`
+
+coala can configure this plugin as:
+
+**Block assignment of issues with labels**: `status/blocked, initiatives/*`
+
+**Block assignment of issues to users**: `gitmate-bot, coala-bot`
+
+**Maximum number of certain labelled issues a member can be assigned**:
+`difficulty/newcomer`: `1`
+
+**Difficulty Level order**:
+`difficulty/newcomer, difficulty/low, difficulty/medium`
+
+### Auto-review merge conflicts
```

Comment:
Change the title to **Notify Merge conflicts in a PR**. There is no review process going on here.